### DISCUSSION TITLE
VS Code command-line tool

### DISCUSSION THREAD

**Discussion by gerry-hornbill** [microsoft/vscode-discussions#1270] [Extension Development QnA] [answered]
Hi, I am wondering if its possible to know, maybe using an IPC or some form, from within a command line tool run within an open terminal window, if the command line tool can determine which file is currently being edited/in focus?

**Comment by alefragnani** [comment, marked as answer]
Hi @gerry-hornbill ,

There is no such feature available via the command line. I suppose that in order to expose such info, you would need to create a new extension that, using VS Code API (I guess `visibleTextEditors` and `tabGroups` are the ones to look for), could provide such info. Then, using some communication protocol, to be reached from outside VS Code.

Hope this helps

**Reply by gerry-hornbill** [reply]
Ok thank you for the clarification and suggestion. I guess making an extension makes sense. 
Gerry
